test_spack.py

```diff
@@ -343,6 +343,28 @@ def test_devbuild_gpu(self):
             self.Run('rm -rf icon-cscs')
 
 
+class IconTestExclaim(TestCase):
```

Comment:
Oh wow, this works. I never designed it to work like this :-)
I designed it to have one test class for each package. But I double checked, it really works correctly.

Reply:
Yes, and I split it in order to not make some test classes so big that all our Srun efforts become worthless.

I started this PR to do it for some other classes too:
https://github.com/C2SM/spack-c2sm/pull/509

Reply:
Back from holiday :-)

Each test class runs a separate srun-command. So if we have few but big test classes we do not gain a lot by using srun.

For many but small test-classes we run much more srun in parallel, therefore we benefit a lot by decreasing total time to run the tests.

Is it clearer now?

Reply:
Each test **case** runs a separate srun command, not each test class.
The tests already run fully parallel.
This code [https://github.com/C2SM/spack-c2sm/blob/master/test_spack.py#L557-L564](https://github.com/C2SM/spack-c2sm/blob/master/test_spack.py#L557-L564) creates a list `asyncMethod` containing all test cases, wrapped into an `async` `startRunCase` function call. This means all the srun calls from within each test case get started in parallel.
Therefore I think splitting the test classes won't speed up anything. Right now the splitting serves just as a grouping of test cases, AFAIK.

Reply:
Hm I observe a different behaviour based on the logfiles that are created for each test of a class.
E.g. an int2lm testrun always creates the logfiles for each test sequentially, hence does not run them in parallel.

But that's not of such big importance. Let's move this discussion to https://github.com/C2SM/spack-c2sm/pull/509 and move on here.

For now the thematical subgrouping for icon makes sense for me.

Reply:
👍